src/Elasticsearch/State/ItemProvider.php

```diff
@@ -26,7 +26,7 @@
 /**
  * Item provider for Elasticsearch.
  *
- * @experimental
+ * @deprecated
```

Comment:
I don't think that this provider should be deprecated, the deprecation was about Metadata Factories the provider will be just fine using the new option if available. .

Reply:
Nope it won’t, because it is using deprecated metadata factories.

Reply:
You can accept both as parameters and deprecate passing the old ones.